tests/test_tpa_pipeline.py

```diff
@@ -127,75 +127,37 @@ def test_get_consent_status(self):
             consent.delete()
             ec_user.delete()
 
-    def test_verify_data_sharing_consent(self):
+    def test_handle_enterprise_logistration(self):
         """
-        Test that we correctly verify consent status.
+        Test that we correctly handle incoming enterprise logistration requests.
         """
         backend = mock.MagicMock(name=None)
         with mock.patch('enterprise.tpa_pipeline.get_ec_for_running_pipeline') as fake_get_ec:
             fake_get_ec.return_value = None
-            assert verify_data_sharing_consent(backend, self.user) is None
+            assert handle_enterprise_logistration(backend, self.user) is None
         with mock.patch('enterprise.tpa_pipeline.get_ec_for_running_pipeline') as fake_get_ec:
-            fake_get_ec.return_value = mock.MagicMock(
-                requests_data_sharing_consent=False
-            )  # pylint: disable=redefined-variable-type
-            assert verify_data_sharing_consent(backend, self.user) is None
+            enterprise_customer = EnterpriseCustomerFactory(
+                enable_data_sharing_consent=False
+            )
+            fake_get_ec.return_value = enterprise_customer  # pylint: disable=redefined-variable-type
+            assert handle_enterprise_logistration(backend, self.user) is None
+            assert EnterpriseCustomerUser.objects.filter(
+                enterprise_customer=enterprise_customer,
+                user_id=self.user.id
+            ).count() == 1
         with mock.patch('enterprise.tpa_pipeline.get_ec_for_running_pipeline') as fake_get_ec:
             fake_get_ec.return_value = self.customer  # pylint: disable=redefined-variable-type
-            assert isinstance(verify_data_sharing_consent(backend, self.user), HttpResponseRedirect)
+            assert isinstance(handle_enterprise_logistration(backend, self.user), HttpResponseRedirect)
             ec_user = EnterpriseCustomerUser.objects.create(
```

Comment:
Nit: This pattern of creating `ec_user` and `consent` objects, and then deleting them when the test is done, is used here and in the test above, [`test_get_consent_status`](https://github.com/open-craft/edx-enterprise/blob/ec78660e56416173846b7873ef852a7ba68b3463/tests/test_tpa_pipeline.py#L119).

Would be worth reducing some code duplication by pulling the creation out into a separate method, which also appends cleanup tasks to delete them when done ([example](https://github.com/edx/edx-platform/blob/6d2468d0dcf782a7561ecc61946146d6fae2762f/common/djangoapps/third_party_auth/tests/specs/test_twitter.py#L21-L28)), like this:

```python
    def create_ec_user_with_consent(self):
        ec_user = EnterpriseCustomerUser.objects.create(
            user_id=self.user.id,  # pylint: disable=no-member
            enterprise_customer=self.customer
        )
        consent = UserDataSharingConsentAudit.objects.create(  # pylint: disable=no-member
            user=ec_user
        )
        self.addCleanup(consent.delete)
        self.addCleanup(ec_user.delete)
        return ec_user, consent
```